experiments/react-native-uifabric/src/components/RadioGroup/RadioButton.tsx

```diff
@@ -0,0 +1,122 @@
+import { View, Text, NativeSyntheticEvent } from 'react-native';
+import {
+  radioButtonName,
+  IRadioButtonType,
+  IRadioButtonProps,
+  IRadioButtonState,
+  IRadioButtonSlotProps,
+  IRadioButtonRenderData
+} from './RadioButton.types';
+import { compose, IUseComposeStyling } from '@uifabricshared/foundation-compose';
+import { filterViewProps } from '../../utilities/RenderHelpers';
+import { ISlots, withSlots } from '@uifabricshared/foundation-composable';
+import { settings } from './RadioButton.settings';
+import { mergeSettings } from '@uifabricshared/foundation-settings';
+import { foregroundColorTokens, textTokens, borderTokens, backgroundColorTokens } from '../../tokens';
+import { useAsPressable } from '../../hooks';
+import { RadioGroupContext } from './RadioGroup';
+
+export const RadioButton = compose<IRadioButtonType>({
+  displayName: radioButtonName,
+
+  usePrepareProps: (userProps: IRadioButtonProps, useStyling: IUseComposeStyling<IRadioButtonType>) => {
+    const { content, buttonKey, disabled, ariaLabel, ...rest } = userProps;
+
+    // Grabs the context information from RadioGroup (currently selected button and client's onChange callback)
+    const info = React.useContext(RadioGroupContext);
+
+    const pressable = useAsPressable({
+      ...rest
+    });
+
+    const state: IRadioButtonState = {
+      ...pressable.state,
+      selected: info.selectedKey === userProps.buttonKey,
+      disabled: disabled || false
+    };
+
+    // Grab the styling information from the userProps, referencing the state as well as the props.
+    const styleProps = useStyling(userProps, (override: string) => state[override] || userProps[override]);
+
+    // This function is called every time a RadioButton gains focus. It does two things:
+    // 1) Calls pressable's onFocus in order to keep track of our state's focus variable. It is dependent on pressable's
+    //    focus variable. Without this, it wouldn't stay updated because we're overriding it's onFocus below for the rootProps.
+    // 2) Selects the currently focused button by calling the RadioGroup's callback function.
+    const onFocusChange = (ev: NativeSyntheticEvent<{}>) => {
+      pressable.props.onFocus && pressable.props.onFocus(ev);
+      // This check is necessary because this func gets called even when a button loses focus (not sure why?) which then calls the client's onChange multiple times
+      if (!state.selected) {
+        info.onButtonSelect(buttonKey);
+      }
+    };
+
+    const accessibilityTraits = {
+      accessibilityRole: 'radio',
+      accessibilityLabel: ariaLabel ? ariaLabel : content,
+      // Why are we able to do this? The radio button that's focused is always going to be the one selected. Therefore, unless it's disabled,
+      // we can always set state=selected for the narrator, because every time the narrator is focused on a button, that's the one selected
+      accessibilityStates: state.disabled ? ['disabled'] : ['selected'],
```

Comment:
You should not report the state as 'selected' just because that's how Narrator appears to behave because every AT gets the same info and we need to report the correct info because we don't know what they do with it.

Moreover, this is still a bug in Narrator, because Narrator can query the selection state of any item before it gets actual focus and reads anything aloud and this is an invalid state.

Reply:
Fixed in newest iteration